src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/TaskTransformer.kt

```diff
@@ -96,6 +105,35 @@ class TaskTransformer(
     outcome = placementApplication.decision?.apiValue,
   )
 
+  private fun getPersonNameFromApplication(
+    application: ApplicationEntity,
+    offenderSummaries: List<PersonSummaryInfoResult>,
+  ): String {
+    val crn = application.crn
+    val offenderSummary = offenderSummaries.first { it.crn == crn }
+    return getNameFromPersonSummaryInfoResult(offenderSummary)
+  }
+
+  private fun getPersonSummaryFromApplication(
+    application: ApplicationEntity,
+    offenderSummaries: List<PersonSummaryInfoResult>,
+  ): PersonSummary {
+    val offenderSummary = offenderSummaries.first { it.crn == application.crn }
+    val personType = when (offenderSummary) {
+      is PersonSummaryInfoResult.Success.Full -> PersonSummaryDiscriminator.fullPersonSummary
+      is PersonSummaryInfoResult.Success.Restricted -> PersonSummaryDiscriminator.restrictedPersonSummary
+      is PersonSummaryInfoResult.NotFound, is PersonSummaryInfoResult.Unknown -> PersonSummaryDiscriminator.unknownPersonSummary
+    }
+
+    val name: String? = if (personType == PersonSummaryDiscriminator.fullPersonSummary) {
+      getNameFromPersonSummaryInfoResult(offenderSummary)
+    } else {
+      null
+    }
+
+    return FullPersonSummary(crn = application.crn, personType = personType, name = name)
```

Comment:
I think we should be returning one of three types here:

* FullPersonSummary
* RestrictedPersonSummary
* UnknownPersonSummary

I think the other 2 will need to be defined in the yml, but neither will have any additional fields other than what is inherited from #/components/schemas/PersonSummary.

This can probably be shifted into the when block on line 122, and what ever that returns can be immediately returned as the function result

Take a look at PersonTransformer.transformModelToPersonApi for how we do this for the non-summary types